static/app/views/explore/hooks/useTimeseries.tsx

```diff
@@ -0,0 +1,137 @@
+  // in the past (say, 5 minutes before the 'real timeseries') - this value is the analogue of 'value'
+  // for the second timeseries in the past.
+  // Alerts use this to calculate a relative change in timeseries with ((value - comparisonValue) / comparisonValue) * 100
+  comparisonValue?: number;
+
+  /** confidence: For sampled datasets - whether there is enough data for the timeseries to be accurate. */
+  confidence?: 'low' | 'high';
+
+  /** sampleCount: For sampled datasets - how many samples were used to calculate this datapoint. */
+  sampleCount?: number;
+
+  /** sampleRate: For sampled datasets - 0.1 represents 9/10 events being deleted due to sampling */
+  sampleRate?: number;
+}
+
+/**
+ * SeriesMeta represents metadata about a single timeseries.
+ *
+ * For example, it might say what type of data is in this timeseries.
+ */
+export interface SeriesMeta {
+  /** valueType: The type of the unit, e.g., 'duration' or 'size' */
+  valueType: string;
+
+  /** interval: The rollup for this time series (TODO: better description) */
+  interval?: number;
+
+  /** isOther: for "top events" timeseries - whether this timeseries it the 'other / rest' category */
+  isOther?: string;
+
+  /** order: for "top events" timeseries - the index of this timeseries (0 is most common) */
+  order?: number;
+
+  /** valueUnit: The unit of this chart, e.g., 'microsecond' */
+  valueUnit?: string;
+}
+
+/**
+ * TimeSeries represents a single timeseries being returned.
+ */
+export interface TimeSeries {
+  /** meta: Metadata about this specific timeseries, like its unit */
+  meta: SeriesMeta;
+
+  /** Values: The actual data points for this timeseries */
+  values: Row[];
+
+  /** yaxis: The displayable y-axis, e.g.: sum(span.duration) */
+  yaxis: string;
+
+  /** groupBy: The requested 'GROUP BY' fields for this timeseries */
+  groupBy?: string[];
+}
+
+export interface StatsMeta {
+  /** dataset: Which dataset was queried for this data */
+  dataset: `${DiscoverDatasets}`;
+
+  /** end: The end epoch time (in milliseconds) for the data */
+  end: number;
+
+  /** start: The start epoch time (in milliseconds) for the data */
+  start: number;
+}
+
+/**
+ * StatsResponse returns zero or more timeseries, and associated global metadata.
+ */
+export interface StatsResponse {
+  /** meta: Metadata about the entire response (like which dataset was used) */
+  meta: StatsMeta;
+
+  /** timeseries: The actual timeseries data */
+  timeseries: TimeSeries[];
+}
+
+export interface UseTimeseriesProps {
+  /** The name of the dataset to target for this request */
+  dataset: `${DiscoverDatasets}`;
+
+  /** The end timestamp for this request */
+  end?: any;
+
+  /** The fields to plot, e.g., count() or message */
+  field?: string[];
+
+  /** The keys to order by, e.g., -count() */
+  orderby?: string[];
+
+  /** The project IDs to target */
+  project?: string[];
+
+  /** The start timestamp for this request */
+  start?: any;
```

Comment:
Would be better to just accept `pageFilters` here instead of start, end and projects, IMO, since you're falling back to `pageFilters` lower down. That way you'll also accept `environments` which is important